src/server/api/qrcode.ts

```diff
@@ -0,0 +1,48 @@
+import Express from 'express'
+import Joi from '@hapi/joi'
+import { createValidator } from 'express-joi-validation'
+
+import { ogUrl } from '../config'
+import createGoQrCode from '../util/qrcode'
+import ImageFormat from '../../shared/util/imageFormat'
+import { isValidShortUrl } from '../../shared/util/validation'
+
+function isValidFormat(format: string): boolean {
+  return (Object as any).values(ImageFormat).includes(format)
+}
+
+const qrCodeRequestSchema = Joi.object({
+  url: Joi.string()
+    .custom((url: string, helpers) => {
+      if (!isValidShortUrl(url)) {
+        return helpers.message({ custom: 'Not a valid short link' })
+      }
+      return url
+    })
+    .required(),
+  format: Joi.string()
+    .custom((format: string, helpers) => {
+      if (!isValidFormat(format)) {
+        return helpers.message({ custom: 'Not a valid format' })
+      }
+      return format
+    })
+    .required(),
+})
+
+const router = Express.Router()
+const validator = createValidator()
+
+router.post('/', validator.body(qrCodeRequestSchema), (req, res) => {
```

Comment:
this should be a GET endpoint because the client is requesting for a resource from the server, and not creating a new resource to be stored on the server.

Reply:
That makes sense. I can make that change. Subsequently, I will also shift the query parameters from the body to the query string